### PR TITLE
Obey lock return values

### PR DESCRIPTION
**Pull request by Sheikah45**

### CODE REVIEW

File: src/main/kotlin/com/faforever/icebreaker/service/SessionService.kt [hunk 1]
```diff
@@ -27,25 +28,34 @@ class SessionService(
...
-            val servers = activeSessionHandlers.flatMap {
-                it.createSession(session.id)
-                it.getIceServersSession(session.id)
+                    return Session(
+                        id = session.id,
+                        servers = servers,
+                    )
+                }
+                else -> {
+                    throw TimeoutException("Unable to acquire game lock for $gameId")
```
Comment (Brutus5000) [inline src/main/kotlin/com/faforever/icebreaker/service/SessionService.kt:50]: If you want to throw an exception, might as well do it in IceSessionRepository

**Comment by Sheikah45** [comment]
I changed it so that the lock only guards the database interactions. This resolved the JDBC connection acquisitions errors based on my local load testing.

However this means that the xirsys channel creation is no longer implicitly locked based on the session id. And despite appearances with the createChannel using put, it is not idempotent and will return an error if the channel already exists.

**Comment by Brutus5000** [comment]
See my other PRs how to read/handle known error codes.

File: src/main/kotlin/com/faforever/icebreaker/persistence/IceSessionEntity.kt [hunk 2]
```diff
@@ -23,22 +25,37 @@ data class IceSessionEntity(
...
-        getEntityManager().createNativeQuery("SELECT RELEASE_LOCK(:lockName)", Boolean::class.java).apply {
+        val lockReleased = getEntityManager().createNativeQuery("SELECT RELEASE_LOCK(:lockName)", Boolean::class.java).apply {
             setParameter("lockName", "game_id_$gameId")
-        }.singleResult
+        }.singleResult as Boolean?
+
+        when (lockReleased) {
+            null -> LOG.warn("No lock exists for $gameId")
+            false -> LOG.warn("Not owner of lock for $gameId")
+            true -> LOG.debug("lock released for $gameId")
```
Comment (Brutus5000) [inline src/main/kotlin/com/faforever/icebreaker/persistence/IceSessionEntity.kt:58]: Uppercase l :P

**Review by Brutus5000 — APPROVED** [review]

**Comment by Sheikah45** [comment]
Just as an FYI this still throws errors for the duplicated channel

**Pull request merged**